firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/datagrams.txt

```diff
@@ -0,0 +1,456 @@
+
+    WHEN("A body with a complete 2-byte header, and a non empty payload buffer consistent with the length field of the header, is altered after it's parsed")
+      THEN("The parse method reports ok status")
+      THEN("After the parse method is called, The value returned by the seq accessor method is equal to the sequence field of the original input_buffer body's header")
+      THEN("The value returned by the length accessor method is equal to the length field of the original input_buffer body's header")
+      THEN("The payload buffer returned from the payload accessor method is equal to the original payload from the body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the Datagram constructor") 
+      THEN("the payload given in the Datagram constructor is independent of the input buffer body")
+      THEN("After the input buffer is altered, The payload buffer returned from the payload accessor method is unchanged")
+
+    WHEN("A body with a complete 2-byte header, and paylaod buffer as '0x00' consistent with the length field of the header is parsed")
+      THEN("The parse method reports ok status")
+      THEN("After the parse method is called, The value returned by the seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The payload returned from the payload accessor method is equal to the payload from the input_buffer body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the Datagram constructor") 
+      THEN("the payload given in the Datagram constructor is independent of the input buffer body")
+
+    WHEN("The parse method is called on a body with a complete 2-byte header, and a non-empty payload buffer where the length field of the header is inconsistent with the actual length of the payload buffer")
+      THEN("The parse method reports ok status")
+      THEN("After the parse method is called, The value returned by the seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The value returned by the length accessor method is not equal to the size of the payload buffer in the input_buffer body")
+      THEN("The payload returned from the payload accessor method is equal to the payload from the input_buffer body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the Datagram constructor") 
+      THEN("the payload given in the Datagram constructor is independent of the input buffer body")
+
+    WHEN("The parse method is called on a body with sequence field of the header inconsistent with sequence given in the constructor of the datagram")
```

Comment:
I think it's better to avoid using the word "inconsistent" here, because there is no concept of data consistency between the sequence number given in the constructor and the sequence number being parsed from the body. Maybe "WHEN the parse method is called on a body whose header's sequence number field has a different value than the sequence number used to initialize the Datagram"